OMEdit/OMEditGUI/Animation/AbstractAnimationWindow.cpp

```diff
@@ -96,6 +98,73 @@ AbstractAnimationWindow::AbstractAnimationWindow(QWidget *pParent)
 }
 
 /*!
+ * \brief AbstractAnimationWindow::showShapePickContextMenu
+ * \param pos
+ */
+void AbstractAnimationWindow::showShapePickContextMenu(const QPoint& pos)
+{
+  QString name = QString::fromStdString(mpViewerWidget->getSelectedShape());
+  //std::cout<<"SHOW CONTEXT "<<name.toStdString()<<" compare "<<QString::compare(name,QString(""))<< std::endl;
+  //the context widget
+  QMenu contextMenu(tr("Context menu"), this);
+  QMenu shapeMenu(name, this);
+  QAction action1("Change Transparency", this);
+  QAction action2("Reset All Shapes", this);
+  //if a shape is picked, we can set it transparent
+  if (0 != QString::compare(name,QString("")))
+  {
+    contextMenu.addMenu(&shapeMenu);
+    shapeMenu.addAction(&action1);
+    connect(&action1, SIGNAL(triggered()), this, SLOT(changeShapeTransparency()));
+  }
+  contextMenu.addAction(&action2);
+  connect(&action2, SIGNAL(triggered()), this, SLOT(removeTransparencyForAllShapes()));
+  connect(&contextMenu, SIGNAL(aboutToHide()), this, SLOT(doSomething()));
+  contextMenu.exec(QWidget::mapToGlobal(pos));
```

Comment:
You need to use `mpViewerWidget->mapToGlobal(pos)` to get the context menu at the correct position.